File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;

/**
 * 品牌接口
 *
 * @author liuweijian
 * @version 2018/11/8.
 */
public interface BrandService {

	List<TbBrand> findAll();

	/**
	 * 品牌分页
	 * @param pageNum 当前页码
	 * @param pageSize 页记录数
	 * @return
	 */
	PageResult findPage(int pageNum, int pageSize);

	/**
	 * 品牌添加
	 * @param brand
	 */
	void add(TbBrand brand);

	/**
	 * 根据ID查找品牌信息
	 * @param id
	 * @return
	 */
	TbBrand findOne(Long id);

	/**
	 * 更新品牌信息
	 * @param brand
	 */
	void update(TbBrand brand);

	/**
	 * 删除品牌
	 * @param ids
	 */
	void delete(Long[] ids);
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
// 品牌服务层
app.service("brandService", function ($http) {
    // 查询品牌列表服务
    this.findAll = function () {
        return $http.get('../brand/findAll.do');
    }

    // 分页条件查询品牌列表服务
    this.search = function(page, size, searchEntity) {
        return $http.post('../brand/search.do?page=' + page + '&size=' + size, searchEntity);
    }

    // 品牌添加服务
    this.add = function (entity) {
        return $http.post('../brand/add.do', entity)
    }

    // 品牌更新服务
    this.update = function (entity) {
        return $http.post('../brand/update.do', entity)
    }

    // 品牌查询服务
    this.findOne = function (id) {
        return $http.get("../../brand/findOne.do?id=" + id);
    }

    //品牌删除服务
    this.dele = function (ids) {
        return $http.get("../brand/delete.do?ids=" + ids);
    }

    //查询品牌下拉列表数据
    this.selectOptionList=function () {
        return $http.get('../brand/selectOptionList.do');
    }
})